core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -48,16 +49,16 @@ object UnitAutomation {
 
     private fun tryGoToRuinAndEncampment(unit: MapUnit): Boolean {
         if (!unit.civInfo.isMajorCiv()) return false // barbs don't have anything to do in ruins
-        val unitDistanceToTiles = unit.movement.getDistanceToTiles()
-        val tileWithRuinOrEncampment = unitDistanceToTiles.keys
+
+        val tileWithRuinOrEncampment = unit.viewableTiles
             .firstOrNull {
                 (
                     (it.improvement != null && it.getTileImprovement()!!.isAncientRuinsEquivalent())
                     || it.improvement == Constants.barbarianEncampment
                 )
                 && unit.movement.canMoveTo(it)
             } ?: return false
-        unit.movement.moveToTile(tileWithRuinOrEncampment)
+        unit.movement.headTowards(unit.movement.getShortestPath(tileWithRuinOrEncampment,true)[0])
```

Comment:
(reposting this because I think I messed up the previous comment)

This could be simplified with unit.movement.headTowards(tileWithRuinOrEncampment) since that already calls unit.movement.getShortestPath(destination) if it takes more than a turn to reach the ruins (only difference being it doesn't avoid damaging terrain but I don't know how much that would make a difference) (also I don't know why this comment says pending...).

Android Studio also says import com.unciv.models.ruleset.Nation is unused.

Reply:
> This could be simplified with `unit.movement.headTowards(tileWithRuinOrEncampment)` since that already calls `unit.movement.getShortestPath(destination)`

If that's the case then alright

> Android Studio also says `import com.unciv.models.ruleset.Nation` is unused.

:+1:

Reply:
I've updated the file to include these fixes